File: packages/shared/src/types/note.ts
export interface Note {
  id: string;
  title: string;
  content: string;
  path: string;
  folder: string;
  tags: string[];
  created: string;
  modified: string;
  pinned: boolean;
}

export interface NoteListItem {
  id: string;
  title: string;
  path: string;
  folder: string;
  tags: string[];
  created: string;
  modified: string;
  pinned: boolean;
  preview: string;
}

export interface Folder {
  name: string;
  path: string;
  children: Folder[];
  noteCount: number;
}

export interface Tag {
  name: string;
  count: number;
}

export interface SearchResult {
  path: string;
  title: string;
  matches: SearchMatch[];
}

export interface SearchMatch {
  line: number;
  content: string;
  start: number;
  end: number;
}

export interface TrashItem {
  id: string;
  title: string;
  original_path: string;
  trash_path: string;
  deleted_at: string;
  preview: string;
}

File: src-tauri/src/models/note.rs
use chrono::Utc;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Note {
    pub id: String,
    pub title: String,
    pub content: String,
    pub path: String,
    pub folder: String,
    pub tags: Vec<String>,
    pub created: String,
    pub modified: String,
    pub pinned: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NoteListItem {
    pub id: String,
    pub title: String,
    pub path: String,
    pub folder: String,
    pub tags: Vec<String>,
    pub created: String,
    pub modified: String,
    pub pinned: bool,
    pub preview: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NoteFrontmatter {
    pub title: String,
    #[serde(default)]
    pub tags: Vec<String>,
    pub created: String,
    pub modified: String,
    #[serde(default)]
    pub pinned: bool,
}

impl NoteFrontmatter {
    pub fn with_title(title: String) -> Self {
        let now = Utc::now().to_rfc3339();
        Self {
            title,
            tags: Vec::new(),
            created: now.clone(),
            modified: now,
            pinned: false,
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SearchResult {
    pub path: String,
    pub title: String,
    pub matches: Vec<SearchMatch>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SearchMatch {
    pub line: usize,
    pub content: String,
    pub start: usize,
    pub end: usize,
}
